xikolo-common/Data/Helper/FetchRequests/AnnouncementHelper+FetchRequests.swift

```diff
@@ -21,10 +21,18 @@ extension AnnouncementHelper {
 
         static var unreadAnnouncements: NSFetchRequest<Announcement> {
             let request: NSFetchRequest<Announcement> = Announcement.fetchRequest()
+            let dateSort = NSSortDescriptor(key: "publishedAt", ascending: false)
+            request.sortDescriptors = [dateSort]
             request.predicate = NSPredicate(format: "visited = %@", NSNumber(booleanLiteral: false))
             return request
         }
 
+        static func courseAnnouncements(forCourseId courseId: String) -> NSFetchRequest<Announcement> {
```

Comment:
I would prefer the following method signature
```swift
static func announcements(forCourse course: Course) -> NSFetchRequest<Announcement>
````

The predicate can be adapted as well.
```swift
NSPredicate(format: "course = %@", course)
```